### PR TITLE
KAT-2144: Rich structured rendering for planning artifacts

### PR DESCRIPTION
**Pull request by gannonh**
## Summary
- add artifact-parser with dedicated parsers for ROADMAP, REQUIREMENTS, DECISIONS, and CONTEXT plus artifact type detection
- add rich planning artifact renderer components (RoadmapView, RequirementsView, DecisionsView, ContextView) and structured/fallback rendering in PlanningPane
- add planning artifact tab navigation (ArtifactTabs) with fixed ordering, active state, per-artifact scroll restore, and unseen update indicators backed by lastViewedPlanningArtifactAtom

## Validation
- bun test apps/desktop/src/renderer/lib/__tests__/artifact-parser.test.ts
- bun run --cwd apps/desktop test
- bun run --cwd apps/desktop typecheck
- bun run --cwd apps/desktop build

## Summary by CodeRabbit

* **New Features**
  * Structured, typed views for roadmaps, requirements, decisions and context with tables, cards and collapsibles.
  * Horizontal artifact tabs with normalized labels and update-dot indicators for unviewed changes.
  * View-tracking per artifact so previously viewed timestamps are remembered; scroll positions are preserved when switching.
  * Amber banner and raw-markdown fallback when structured parsing fails.

* **Bug Fixes**
  * Preserve current active artifact selection when appropriate instead of always overwriting.

* **Tests**
  * New parser test suite validating artifact detection and parsing.

<h3>Greptile Summary</h3>

This PR implements structured rendering for four planning artifact types (ROADMAP, REQUIREMENTS, DECISIONS, CONTEXT) in the Kata Desktop right pane, replacing raw markdown rendering with dedicated view components.

**Key changes:**
- **`artifact-parser.ts`** — New library with `detectArtifactType`, `parseRoadmap`, `parseRequirements`, `parseDecisions`, `parseContext`, and `listTopLevelSections`. Well-tested with representative markdown fixtures.
- **`PlanningPane.tsx`** — Adds tab navigation (via new `ArtifactTabs`), per-artifact scroll restore using a `useRef`-backed map, "unseen update" indicator computation, and structured-vs-fallback rendering dispatch.
- **`ArtifactTabs.tsx`** — New component rendering sorted, labeled tabs with a dot indicator for unviewed updates.
- **View components** (`RoadmapView`, `RequirementsView`, `DecisionsView`, `ContextView`) — Each renders its parsed structure with Tailwind styling; `DecisionsView` adds sortable columns.
- **`planning.ts`** — Adds `lastViewedPlanningArtifactAtom` / `markPlanningArtifactViewedAtom` for tracking unseen updates, and changes `applyPlanningArtifactAtom` to preserve the user's currently active tab instead of always switching to the latest incoming artifact.
- **`types.ts`** — New shared type definitions for all parsed artifact structures.

**Issues found:**
- The "last viewed" tracking atom is keyed by `artifact.title` rather than `artifact.artifactKey`, which can cause incorrect dot-indicator state when two artifacts from different scopes (project vs. issue) share the same title string.
- `RoadmapView` uses criterion text as a React list key, which can produce duplicate-key warnings if two success criteria share identical text.
- `ContextView` uses a `level-heading` composite key that can collide when the same heading appears at the same level more than once in a document.

<h3>Confidence Score: 4/5</h3>

Safe to merge with one title-keyed tracking bug that can cause incorrect dot indicators for same-name multi-scope artifacts.

The parser, view components, and atom logic are solid and well-tested. The main concern is that lastViewedPlanningArtifactAtom and hasUnviewedUpdatesByTitle both key on artifact.title rather than artifact.artifactKey, which can produce incorrect unseen update state when project-scope and issue-scope artifacts share the same title string. The two React key fragility issues in RoadmapView and ContextView are low-impact style concerns. No runtime crashes, no data loss, no security issues.

apps/desktop/src/renderer/atoms/planning.ts and the corresponding hasUnviewedUpdatesByTitle computation in PlanningPane.tsx — the title-based tracking key should be reviewed against multi-scope artifact scenarios.

<details><summary><h3>Important Files Changed</h3></summary>

| Filename | Overview |
|----------|----------|
| apps/desktop/src/renderer/lib/artifact-parser.ts | New parsing library with clean section extraction, regex-based slice/requirement/decision parsing, and safe null returns; logic is correct and well-tested. |
| apps/desktop/src/renderer/atoms/planning.ts | Adds lastViewedPlanningArtifactAtom (keyed by title, not artifactKey) and a write-only marker atom; applyPlanningArtifactAtom now preserves active tab selection correctly. |
| apps/desktop/src/renderer/components/planning/PlanningPane.tsx | Orchestrates tabs, scroll restore, unseen-update tracking, and structured/fallback dispatch; the title-keyed unviewed tracking inherits the same scope-collision risk from the atom. |
| apps/desktop/src/renderer/components/planning/ArtifactTabs.tsx | Fixed-order tab component with dot indicator; sort and label logic are correct. |
| apps/desktop/src/renderer/components/planning/RoadmapView.tsx | Renders slice cards with checkboxes, risk badges, and dependency tags; uses criterion text as React key, which can produce duplicate-key warnings. |
| apps/desktop/src/renderer/components/planning/ContextView.tsx | Collapsible section renderer backed by Markdown; composite level-heading key can collide for duplicate headings at the same depth. |
| apps/desktop/src/renderer/components/planning/DecisionsView.tsx | Sortable decision table with inline rationale sub-row; sort logic and badge rendering are correct. |
| apps/desktop/src/renderer/components/planning/RequirementsView.tsx | Grouped requirements table with status badge and traceability summary; logic is correct. |
| apps/desktop/src/shared/types.ts | Adds ArtifactType union and all ParsedXxx interfaces; new types are clean and match parser outputs. |
| apps/desktop/src/renderer/lib/__tests__/artifact-parser.test.ts | Tests cover happy-path parsing and malformed-input fallback for all four parsers; fixture data is realistic and edge cases are exercised. |

</details>

</details>

<h3>Flowchart</h3>

```mermaid
%%{init: {'theme': 'neutral'}}%%
flowchart TD
    A[PlanningArtifact arrives via IPC / listArtifacts] --> B[applyPlanningArtifactAtom]
    B --> C{activeArtifact already set?}
    C -- No --> D[Set activePlanningArtifactAtom to incoming artifact]
    C -- Yes --> E[Keep current active tab]
    D & E --> F[planningArtifactsAtom updated]
    F --> G[PlanningPane renders]
    G --> H[detectArtifactType on title]
    H --> I{Known type?}
    I -- roadmap --> J[parseRoadmap to RoadmapView]
    I -- requirements --> K[parseRequirements to RequirementsView]
    I -- decisions --> L[parseDecisions to DecisionsView]
    I -- context --> M[parseContext to ContextView]
    I -- unknown or parse failed --> N[Markdown fallback]
    G --> O[ArtifactTabs renders sorted tabs]
    O --> P{hasUnviewedUpdatesByTitle computed from lastViewedByTitle}
    P -- unseen --> Q[dot indicator shown]
    P -- seen --> R[no indicator]
    G --> S[markPlanningArtifactViewed on activeArtifact change]
    S --> T[lastViewedPlanningArtifactAtom updated by title]
```

<details><summary>Prompt To Fix All With AI</summary>

`````markdown
This is a comment left during a code review.
Path: apps/desktop/src/renderer/atoms/planning.ts
Line: 35-43

Comment:
**Unviewed-update tracking keyed by title, not artifactKey**

`lastViewedPlanningArtifactAtom` uses `artifact.title` as the map key. `buildPlanningArtifactKey` (in `types.ts`) produces distinct keys for the same title at different scopes — e.g. `project:abc:M002-ROADMAP` vs `issue:xyz:M002-ROADMAP`. Because both artifacts share the same `.title` string, marking the project-scope artifact as viewed will suppress the dot indicator on the issue-scope artifact (and vice versa), even though the user may never have opened it.

Using `artifactKey` throughout (atom, marker, and `hasUnviewedUpdatesByTitle` computation in `PlanningPane.tsx`) would eliminate the collision. The `markPlanningArtifactViewed` call at line 117 of `PlanningPane.tsx` would pass `artifactKey` instead of `title`, and the `hasUnviewedUpdatesByTitle` reduce at line 179 would key on `artifact.artifactKey` as well.

How can I resolve this? If you propose a fix, please make it concise.

---

This is a comment left during a code review.
Path: apps/desktop/src/renderer/components/planning/RoadmapView.tsx
Line: 39-41

Comment:
**Fragile React key using criterion text**

Using the criterion string as a `key` will trigger a React duplicate-key warning if two success criteria happen to share identical text. An index-based key (e.g. using the second map callback argument) is safer here since the list order is stable and parsed sequentially from markdown.

How can I resolve this? If you propose a fix, please make it concise.

---

This is a comment left during a code review.
Path: apps/desktop/src/renderer/components/planning/ContextView.tsx
Line: 14-16

Comment:
**Composite key can collide for repeated headings**

The key `${section.level}-${section.heading}` will produce the same string if a document repeats the same heading at the same depth (e.g. two `## Overview` sections). Because sections are parsed in strict document order, using the array index (via the second `.map()` callback argument) as the key is both unique and stable.

How can I resolve this? If you propose a fix, please make it concise.
`````

</details>

<sub>Reviews (1): Last reviewed commit: ["feat(desktop): KAT-2145 KAT-2146 add str..."](https://github.com/gannonh/kata/commit/46bef02e73eae9fcfce5ad303e3d7370ed6d4960) | [Re-trigger Greptile](https://app.greptile.com/api/retrigger?id=27194495)</sub>

> Greptile also left **3 inline comments** on this PR.

### CODE REVIEW

File: apps/desktop/src/renderer/atoms/planning.ts [hunk 1]
```diff
@@ -30,9 +30,20 @@ export const rightPaneModeAtom = atomWithStorage<'planning' | 'default'>(
...
 export const planningLoadingAtom = atom<boolean>(false)
 export const planningErrorAtom = atom<string | null>(null)
+export const lastViewedPlanningArtifactAtom = atom<Record<string, string>>({})
+
+export const markPlanningArtifactViewedAtom = atom(
+  null,
+  (get, set, artifact: { title: string; updatedAt: string }) => {
+    set(lastViewedPlanningArtifactAtom, {
+      ...get(lastViewedPlanningArtifactAtom),
+      [artifact.title]: artifact.updatedAt,
```
[the remark this answers is not in the document]
Comment (gannonh) [reply apps/desktop/src/renderer/atoms/planning.ts]: Great catch — fixed in `a802f628`.

I changed viewed-artifact tracking to key by `artifactKey` instead of `title` so duplicate titles across scopes/projects/issues no longer collide:

- `markPlanningArtifactViewedAtom` now stores `lastViewedPlanningArtifactAtom[artifactKey]`
- `PlanningPane` computes unseen-update state by `artifact.artifactKey`
- `ArtifactTabs` now reads `hasUnviewedUpdatesByKey`

This preserves independent "updated" dot behavior for artifacts that share a title.

File: apps/desktop/src/renderer/lib/artifact-parser.ts [hunk 2]
```diff
@@ -0,0 +1,448 @@
...
+
+function parseRevisableCell(value: string): { value: boolean | null; condition: string | null } {
+  const normalized = value.toLowerCase()
+
+  if (normalized.startsWith('no')) {
+    return { value: false, condition: null }
+  }
+
+  if (normalized.startsWith('yes')) {
+    const condition = value.split(/[—-]/).slice(1).join('—').trim()
```
[the remark this answers is not in the document]
Comment (gannonh) [reply apps/desktop/src/renderer/lib/artifact-parser.ts]: Addressed in `a802f628`.

`parseRevisableCell` now only splits when there is an explicit `Yes` delimiter (`Yes - ...` or `Yes — ...`) and preserves any hyphens/em dashes in the condition text itself.

I also added an assertion in `artifact-parser.test.ts` for `D010` to verify `revisableCondition` remains `if auto-switch is disruptive`.

File: apps/desktop/src/renderer/atoms/planning.ts [hunk 3]
```diff
@@ -30,9 +30,20 @@ export const rightPaneModeAtom = atomWithStorage<'planning' | 'default'>(
...
+    })
+  },
+)
```
[the remark this answers is not in the document]
Comment (gannonh) [reply apps/desktop/src/renderer/atoms/planning.ts:43]: Fixed — this is now keyed by `artifactKey` end-to-end (see `a802f628`).

- `markPlanningArtifactViewedAtom` stores by `artifactKey`
- `PlanningPane` passes `artifactKey` when marking viewed
- unseen-update map is computed and consumed by `artifact.artifactKey`

So project/issue artifacts with the same title no longer collide.

File: apps/desktop/src/renderer/components/planning/RoadmapView.tsx [hunk 4]
```diff
@@ -0,0 +1,115 @@
...
+
+      {roadmap.successCriteria.length > 0 ? (
+        <section className="rounded-lg border border-border bg-background px-3 py-2">
+          <p className="text-xs font-semibold tracking-wide uppercase text-muted-foreground">
+            Success Criteria
+          </p>
+          <ul className="mt-2 list-disc space-y-1 pl-4 text-sm">
+            {roadmap.successCriteria.map((criterion) => (
+              <li key={criterion}>{criterion}</li>
+            ))}
```
[the remark this answers is not in the document]
Comment (gannonh) [reply apps/desktop/src/renderer/components/planning/RoadmapView.tsx]: Good call — fixed in `c96e1afa`.

I switched the success-criteria list key to a stable index-based key (`criterion-${index}`), which avoids duplicate-key warnings when criterion text repeats.

File: apps/desktop/src/renderer/components/planning/ContextView.tsx [hunk 5]
```diff
@@ -0,0 +1,46 @@
...
+export interface ContextViewProps {
+  context: ParsedContext
+}
+
+export function ContextView({ context }: ContextViewProps) {
+  return (
+    <div className="space-y-3">
+      {context.sections.map((section) => (
+        <Collapsible
+          key={`${section.level}-${section.heading}`}
```
[the remark this answers is not in the document]
Comment (gannonh) [reply apps/desktop/src/renderer/components/planning/ContextView.tsx]: Fixed in `c96e1afa`.

I updated the section map key to an index-based key (`context-section-${index}`), which is unique and stable for parsed document order even if headings repeat at the same level.

File: apps/desktop/src/renderer/atoms/planning.ts [hunk 6]
```diff
@@ -30,9 +30,20 @@ export const rightPaneModeAtom = atomWithStorage<'planning' | 'default'>(
...
+export const lastViewedPlanningArtifactAtom = atom<Record<string, string>>({})
+
+export const markPlanningArtifactViewedAtom = atom(
+  null,
+  (get, set, artifact: { title: string; updatedAt: string }) => {
+    set(lastViewedPlanningArtifactAtom, {
+      ...get(lastViewedPlanningArtifactAtom),
+      [artifact.title]: artifact.updatedAt,
+    })
+  },
```
[the remark this answers is not in the document]
Comment (gannonh) [reply apps/desktop/src/renderer/atoms/planning.ts:42]: Handled — this was already addressed in `a802f628` and remains in the branch.

`lastViewedPlanningArtifactAtom`/`markPlanningArtifactViewedAtom` and unseen-update computation are all keyed by `artifactKey` now, not `title`, so same-title artifacts in different scopes no longer collide.

File: apps/desktop/src/renderer/components/planning/ArtifactTabs.tsx [hunk 7]
```diff
@@ -0,0 +1,86 @@
...
+            className={cn(
+              'relative shrink-0 rounded-t-md border border-transparent px-3 py-1.5 text-xs font-medium text-muted-foreground transition-colors hover:bg-muted/50 hover:text-foreground',
+              isActive && 'border-border border-b-background bg-background text-foreground',
+            )}
+          >
+            <span>{formatArtifactTitle(artifact.title)}</span>
+            {hasUnviewedUpdate ? (
+              <span className="absolute top-1 right-1 size-1.5 rounded-full bg-primary" aria-label="updated" />
+            ) : null}
+          </button>
```
[the remark this answers is not in the document]
Comment (gannonh) [reply apps/desktop/src/renderer/components/planning/ArtifactTabs.tsx]: Implemented in `51dc85dc`.

I added tab semantics in `ArtifactTabs` and linked panel semantics in `PlanningPane`:

- tab strip: `role="tablist"`
- each tab button: `role="tab"`, `aria-selected`, `id`, `aria-controls`, and managed `tabIndex`
- content container: `role="tabpanel"`, matching `id`, and `aria-labelledby`

This preserves existing behavior while making the tab model accessible to assistive tech.

File: apps/desktop/src/renderer/components/planning/ArtifactTabs.tsx [hunk 8]
```diff
@@ -0,0 +1,86 @@
...
+
+  if (detectedType === 'decisions') {
+    return 'Decisions'
+  }
+
+  if (detectedType === 'context') {
+    return 'Context'
+  }
+
+  return title
```
[the remark this answers is not in the document]
Comment (gannonh) [reply apps/desktop/src/renderer/components/planning/ArtifactTabs.tsx]: Addressed in `51dc85dc`.

Tab labels now keep identity when there are duplicates of the same artifact type:

- `ArtifactTabs` computes a per-type frequency map.
- `formatArtifactTitle` only shortens to generic labels (`Roadmap`, `Requirements`, etc.) when that type appears once.
- When multiple artifacts share a type, it preserves the original artifact title for disambiguation.

File: apps/desktop/src/renderer/components/planning/ContextView.tsx [hunk 9]
```diff
@@ -0,0 +1,46 @@
...
+
+          <CollapsibleContent className="border-t border-border bg-background px-3 py-2">
+            {section.content ? (
+              <Markdown mode="full" className="text-sm leading-relaxed">
+                {section.content}
+              </Markdown>
+            ) : (
+              <p className="text-xs text-muted-foreground">No content</p>
+            )}
+          </CollapsibleContent>
```
[the remark this answers is not in the document]
Comment (gannonh) [reply apps/desktop/src/renderer/components/planning/ContextView.tsx]: Implemented in `51dc85dc`.

`ContextView` now groups level-3 sections under the preceding level-2 parent so hierarchy is preserved in rendering:

- parent section content renders first
- nested level-3 sections render inside the parent collapsible
- `No content` is shown only when a section has neither direct content nor nested children

This keeps nested context sections together and ensures collapsing the parent hides its children.
